### PR TITLE
bugfix(uniqueSelector): guard against NPE's from transient nodes

### PR DESCRIPTION
**Pull request by dabrady**
*Contributes to [DG-138](https://prodperfect.atlassian.net/browse/DG-138)*
*Follow-up to #69,  #72*

[A partner reported an error](https://prodperfect.slack.com/archives/CGQMNF3SP/p1624480773076900) after we introduced the "unique selector" to our tracking event payloads. My investigation pointed fingers at events stemming from "transient" nodes, and in particular this scenario:
1. Node is in DOM
2. Event triggered on node
3. Node removed from DOM before event handling completes

This may seem like an edge-case, and indeed it is; but it's a common one, given the level of dynamism in modern web applications. Unfortunately for us, the `unique-selector` library that we chose to use does not guard against being called on elements that are not part of a document, and so we have to do our best to guard against it ourselves; that's where this PR comes in.

I've introduced a last-minute check that the element is part of the document before calling into the `unique-selector` library, which should catch most of these edge-cases.

Furthermore, I've wrapped the call in some error-handling that will squash the null-pointer exceptions thrown by the library in the even more unlikely-but-possible case where the node is removed _after_ we call the library but _before_ the library does the thing that becomes unsafe when the element is not part of a document. I've chosen to squash them because in this case, we can't do anything about them and we also shouldn't let them bubble up to our clients' production applications.

### CODE REVIEW

File: lib/helpers/getBasicDomNodeProfile.js [hunk 1]
```diff
@@ -13,30 +13,43 @@ function getAttr(el, name) {
...
-    all_attrs: allNodeAttributes,
+    all_attrs: {
+      ...getDomNodeAttributes(el, EXCLUDE_VALUE_REGEX),
+      /**
+       * NOTE(dabrady) We are putting this in the `all_attrs` blob out of a variety of
+       * concerns, but primarily because it will let us evalutate our use-case for this
+       * data faster and with minimal effort.
+       */
+      // Don't include uniqueSelector if it couldn't be calculated.
+      unique_selector: generateUniqueSelector(el) || undefined,
```
Comment (dabrady) [inline lib/helpers/getBasicDomNodeProfile.js:51]: As part of this, I simplified the hotfix I made earlier around ensuring the `unique_selector` field is not added to `all_attrs` unless it has a value.

When a JS object is serialized into JSON and sent over the wire, `undefined` properties are ignored. You can easily confirm this by smoke testing the JSON serializer directly in a console:
```javascript
> JSON.stringify({ a: 42, b: null, c: undefined, d: false })
< "{\"a\":42,\"b\":null,\"d\":false}"
```

Comment (pevner-p2) [reply lib/helpers/getBasicDomNodeProfile.js:51]: This is very clever.

**Review by pevner-p2 — APPROVED** [review]
Though I don't know the nuances of JS concurrency, this appears reasonable to me.

**Review by rhymiz — APPROVED** [review]

**Comment by dabrady** [comment]
Associated Semaphore build is here: https://semaphoreci.com/prodperfect/prodperfect-keen-tracking-js/branches/master/builds/1545

**Pull request merged**